pkgs/os-specific/darwin/apple-source-releases/bootstrap_cmds/package.nix

```diff
@@ -0,0 +1,81 @@
+{
+  lib,
+  apple-sdk,
+  bison,
+  buildPackages,
+  fetchFromGitHub,
+  flex,
+  mkAppleDerivation,
+  stdenv,
+  stdenvNoCC,
+  xcbuild,
+}:
+
+let
+  Libc = apple-sdk.sourceRelease "Libc";
+
+  # Older versions of xnu are missing required headers. The one for the 11.3 SDK doesn’t define a needed function
+  # that was added in 11.3. This version of xnu has everything that’s needed.
+  xnu = fetchFromGitHub {
+    owner = "apple-oss-distributions";
+    repo = "xnu";
+    rev = "xnu-8019.41.5";
+    hash = "sha256-CqKTYu10mTYXbEeZwDoDMw3+jKVLs/KSiAHU7nb0Lks=";
+  };
+
+  privateHeaders = stdenvNoCC.mkDerivation {
+    name = "adv_cmds-deps-private-headers";
+
+    buildCommand = ''
+      install -D -t "$out/include/os" \
+        '${Libc}/os/assumes.h' \
+        '${xnu}/libkern/os/base_private.h'
+    '';
+  };
+
+in
+mkAppleDerivation {
+  releaseName = "bootstrap_cmds";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  xcodeHash = "sha256-N28WLkFo8fXiQqqpmRmOBE3BzqXHIy94fhZIxEkmOw4=";
+  xcodeProject = "mig.xcodeproj";
+
+  postPatch = ''
+    # Fix the name to something Meson will like.
+    substituteInPlace migcom.tproj/lexxer.l \
+      --replace-fail 'y.tab.h' 'parser.tab.h'
+  '';
+
+  env.NIX_CFLAGS_COMPILE = "-I${privateHeaders}/include";
+
+  nativeBuildInputs = [
+    bison
+    flex
+  ];
+
+  postInstall = ''
+    mv "$out/bin/mig.sh" "$out/bin/mig"
+    chmod a+x "$out/bin/mig"
+    patchShebangs --build "$out/bin/mig"
+
+    substituteInPlace "$out/bin/mig" \
+      --replace-fail 'arch=`/usr/bin/arch`' 'arch=${stdenv.targetPlatform.darwinArch}' \
+      --replace-fail '/usr/bin/mktemp' '${lib.getBin buildPackages.coreutils}/bin/mktemp' \
+      --replace-fail '/bin/rmdir "''${WORKTMP}"' "" \
+      --replace-fail '/usr/bin/xcrun' '${buildPackages.xcbuild.xcrun}/bin/xcrun'
```

Comment:
It doesn’t work on outputs because they don’t have `meta.mainProgram` defined. The logical `mainProgram` for xcbuild would be `xcodebuild` not `xcrun`, so it probably has to do this.

```
nix-repl> lib.getExe xcbuild.xcrun
trace: evaluation warning: getExe: Package "xcbuild-0.1.1-unstable-2019-11-20" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".
```

Reply:
Sounds like a job for structured attrs. 🤔